docs/source/reference/check.rst

```diff
@@ -55,3 +56,54 @@ Chainer provides some utilities for testing its functions.
    :nosignatures:
 
    chainer.testing.unary_math_function_unittest
+   chainer.testing.save_and_load
+   chainer.testing.save_and_load_hdf5
+   chainer.testing.save_and_load_npz
+   chainer.testing.get_trainer_with_mock_updater
+
```

Comment:
I think these are mainly for Trainer extensions, so it shouldn't be under `Function testing utilities` catgory.
How about adding a new category for these functions?